### PR TITLE
Render teacher_markdown clientside

### PR DESCRIPTION
**Pull request by Hamms**
So we can render it with our SafeMarkdown component.

Prerequisite to opening this field up for translations.

Because this is switching the renderer from redcarpet to remark, it also requires content updates for many of the affected levels; that work is being tracked [here](https://docs.google.com/spreadsheets/d/1p-KP5RMQn3d-aNuagCCyGKyUfC7L8893gPoT2e3fCmA/edit#gid=153934544).

## Links

- [jira](https://codedotorg.atlassian.net/browse/FND-730)
- [content updates](https://docs.google.com/spreadsheets/d/1p-KP5RMQn3d-aNuagCCyGKyUfC7L8893gPoT2e3fCmA/edit#gid=153934544)

## Testing story

As with other renderer switches, the testing pattern for this is to:

- Output all affected markdown to a `markdown.json` file
- Render all markdown with Redcarpet, output to `redcarpet.json`
- Render all markdown with Remark, output to `remark.json`
- Use https://github.com/Hamms/html-equivalency to compare `redcarpet.json` and `remark.json` and identify the levels for which there is an html change

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/views/levels/_teacher_markdown.html.haml [hunk 1]
```diff
@@ -1,7 +1,12 @@
+- content_for(:head) do
+  - # only add a single script time, even if this view is included multiple times in the page
```
Comment (davidsbailey) [inline dashboard/app/views/levels/_teacher_markdown.html.haml]:
```suggestion
  - # only add a single script, even if this view is included multiple times in the page
```

File: apps/Gruntfile.js [hunk 2]
```diff
@@ -542,6 +542,7 @@ describe('entry tests', () => {
     'levels/_bubble_choice':
       './src/sites/studio/pages/levels/_bubble_choice.js',
     'levels/_content': './src/sites/studio/pages/levels/_content.js',
+    'levels/_teacher_markdown': './src/sites/studio/pages/levels/_teacher_markdown.js',
```
Comment (davidsbailey) [inline apps/Gruntfile.js]: I hate to make style comments now that we have prettier, but can you please preserve the alphabetical order?

File: dashboard/app/views/levels/instructions.haml [hunk 3]
```diff
@@ -10,23 +10,6 @@
       border: 1px dashed black;
       padding: 10px;
     }
-    .teacher_markdown {
```
Comment (davidsbailey) [inline dashboard/app/views/levels/instructions.haml:13]: nice cleanup!

**Review by davidsbailey — APPROVED** [review]
Code changes look good, but can you please remind me, why is it important to switch to client-side rendering via SafeMarkdown? is the important part to use remark rather than redcarpet, and can we not render remark server-side?

**Comment by Hamms** [comment]
The important part is to use a renderer that can detect and remove "unsafe" markup; we currently have a solution for this using remark and [hast-util-sanitize](https://github.com/syntax-tree/hast-util-sanitize) and do not have a solution for redcarpet, nor do we have a way to render remark serverside.

**Comment by davidsbailey** [comment]
> The important part is to use a renderer that can detect and remove "unsafe" markup; we currently have a solution for this using remark and [hast-util-sanitize](https://github.com/syntax-tree/hast-util-sanitize) and do not have a solution for redcarpet, nor do we have a way to render remark serverside.

Thanks @Hamms . what categories of content need detection/removal of "unsafe" markup?

**Comment by Hamms** [comment]
Any content that is externally sourced; in this case, sourced from translators.

**Comment by Hamms** [comment]
All affected levels have been updated (https://docs.google.com/spreadsheets/d/1p-KP5RMQn3d-aNuagCCyGKyUfC7L8893gPoT2e3fCmA/edit#gid=153934544), and I've confirmed that no new inconsistencies were added in the last two weeks, so this is now safe to merge!

**Pull request merged**